File: backend/routes/notifications.py
from fastapi import APIRouter, Depends
from database import db
from auth_helpers import get_current_user

router = APIRouter(prefix="/api/notifications", tags=["notifications"])

@router.get("")
async def get_notifications(user=Depends(get_current_user)):
    return await db.notifications.find({"user_id": user["_id"]}, {"_id": 0}).sort("created_at", -1).to_list(50)

@router.get("/unread-count")
async def unread_count(user=Depends(get_current_user)):
    count = await db.notifications.count_documents({"user_id": user["_id"], "read": False})
    return {"count": count}

@router.post("/mark-read")
async def mark_all_read(user=Depends(get_current_user)):
    await db.notifications.update_many({"user_id": user["_id"], "read": False}, {"$set": {"read": True}})
    return {"message": "All marked as read"}

@router.post("/{notification_id}/read")
async def mark_one_read(notification_id: str, user=Depends(get_current_user)):
    await db.notifications.update_one({"id": notification_id}, {"$set": {"read": True}})
    return {"message": "Marked as read"}

File: backend/routes/notifications.js
const express = require('express');
const Notification = require('../models/Notification');
const auth = require('../middleware/auth');

const router = express.Router();

// GET /api/notifications
router.get('/', auth, async (req, res) => {
  try {
    const notifs = await Notification.find({ user_id: req.user._id })
      .select('-_id -__v').sort({ created_at: -1 }).limit(50).lean();
    res.json(notifs);
  } catch (err) {
    res.status(500).json({ detail: err.message });
  }
});

// GET /api/notifications/unread-count
router.get('/unread-count', auth, async (req, res) => {
  try {
    const count = await Notification.countDocuments({ user_id: req.user._id, read: false });
    res.json({ count });
  } catch (err) {
    res.status(500).json({ detail: err.message });
  }
});

// POST /api/notifications/mark-read
router.post('/mark-read', auth, async (req, res) => {
  try {
    await Notification.updateMany({ user_id: req.user._id, read: false }, { $set: { read: true } });
    res.json({ message: 'All marked as read' });
  } catch (err) {
    res.status(500).json({ detail: err.message });
  }
});

// POST /api/notifications/:id/read
router.post('/:id/read', auth, async (req, res) => {
  try {
    await Notification.updateOne({ id: req.params.id }, { $set: { read: true } });
    res.json({ message: 'Marked as read' });
  } catch (err) {
    res.status(500).json({ detail: err.message });
  }
});

module.exports = router;
